### PR TITLE
chore: add issue template

### PR DESCRIPTION
**Pull request by yizha1**
Add issue template for `notation-go`, which is copied from `notation` repo.

Signed-off-by: Yi Zha <yizha1@microsoft.com>

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-or-issue.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    attributes:
+      label: "How can we reproduce it?"
+      description: "Detailed steps to reproduce the behavior. Commands and their outputs are always helpful. If the bug is in a library, code snippets work as well."
+  - type: textarea
+    id: environment
+    validations:
+      required: true
+    attributes:
+      label: Describe your environment 
+      description: "Notation installation method (e.g. wget, curl, brew, apt-get, yum, chocolate, MSI) if applicable / OS version / Shell type (e.g. zsh, bash, cmd.exe, Bash on Windows) / Golang version if applicable"
```
Comment (priteshbandi) [inline .github/ISSUE_TEMPLATE/bug-or-issue.yaml:49]: Not applicable for notation-go
```suggestion
      description: "OS and Golang version"
```

File: .github/ISSUE_TEMPLATE/feature-request.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    id: context
+    validations:
+      required: false
+    attributes:
+      label: "Any additional context?"
+      description: "Add any other context or screenshots about the feature request here."
+  - type: markdown
+    attributes:
+      value: |
+        If you want to contribute to this project, we will be happy to guide you through out contribution process especially when you already have a good proposal or understanding of how to improve the functionality. Join us at https://slack.cncf.io/ and choose #notary-project channel.
```
Comment (Two-Hearts) [inline .github/ISSUE_TEMPLATE/feature-request.yaml:53]: nit:
```suggestion
        If you want to contribute to this project, we will be happy to guide you through the contribution process especially when you already have a good proposal or understanding of how to improve the functionality. Join us at https://slack.cncf.io/ and choose #notary-project channel.
```

**Review by priteshbandi — DISMISSED** [review]
LGTM

**Review by Two-Hearts — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Pull request merged**